edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -15,10 +15,24 @@ public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
+        const string ImageUpstreamRegexPattern = @"^\$(?<host>.*?)(?<path>:.*)";
```

Comment:
I don't have context here, would you please share the design spec?

Reply:
Ok, I'll add a comment too.